wireapp/kalium
# fix: wrong handling of conversation access role (WPB-10746)

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10746" title="WPB-10746" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-10746</a>  [Android] Mismatch on enabling guests and services between web and android
  </td></table>
  <br />
 


----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

When creating a group with disabled guests, we weren't properly mapping the response fields for access roles (current and deprecated value)

### Causes (Optional)

We forgot to add the extra deprecated field in other places in the code.

### Solutions

Map down `access_role` when available or try to get `access_role_v2` for earlier and current versions.

### Testing

#### Test Coverage (Optional)


#### How to Test

### API v2
Fresh login ->
- Verify `access_role_list` for conversation contains correct values
- Create a group with DISABLED GUESTS (only : [TEAM_MEMBER, SERVICE])
- Create a group with ENABLED GUESTS (and services)

### Api v3+ (latest)
Fresh login ->
- Verify `access_role_list` for conversation contains correct values
- Create a group with DISABLED GUESTS (only : [TEAM_MEMBER, SERVICE])
- Create a group with ENABLED GUESTS (and services)

----------------
For App Inspection query you can run this on your db to check it out:
```
SELECT qualified_id, name, access_role_list, access_list FROM Conversation;
or
SELECT qualified_id, name, access_role_list, access_list FROM Conversation WHERE name LIKE '%conversation_name_you_created%';
```
